utxo/src/utxo_impl/test.rs

```diff
@@ -420,10 +420,10 @@ fn derive_cache_test() {
     let mut cache = UtxosCache::default();
 
     let (utxo, outpoint_1) = create_utxo(10);
-    assert!(cache.add_utxo(utxo, &outpoint_1, false).is_ok());
+    assert!(cache.add_utxo(&outpoint_1, utxo, false).is_ok());
```

Comment:
It should be considered changing the last argument of `add_utxo` from a bool to a two-state enum so it's clearer what's going on in the call sites.

Reply:
I added a TODO on that.